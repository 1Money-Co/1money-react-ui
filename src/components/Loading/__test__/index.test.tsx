import 'jsdom-global/register';
import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import sinon from 'sinon';
import { Loading } from '../index';

configure({ adapter: new Adapter() });

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed') ||
    message.includes('React does not recognize') ||
    message.includes('should not be null') ||
    message.includes('A component is changing an uncontrolled input to be controlled')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Loading', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Loading />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('simulate events', async () => {
    const onClick = sinon.spy(); 
    const wrapper = mount(
      <Loading onClick={onClick} />
    );
    wrapper.find('div').simulate('click');
    expect(onClick.called).toBe(true);
  });
});
