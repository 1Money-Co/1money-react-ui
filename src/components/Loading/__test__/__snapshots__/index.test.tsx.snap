// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
LoadedCheerio {
  "0": <div
    aria-busy="true"
    class="usd1-react-ui-loading p-progress-spinner"
    data-pc-name="progressspinner"
    data-pc-section="root"
    role="progressbar"
  >
    <svg
      class="p-progress-spinner-svg"
      data-pc-section="spinner"
      style="animation-duration:2s"
      viewBox="25 25 50 50"
    >
      <circle
        class="p-progress-spinner-circle"
        cx="50"
        cy="50"
        data-pc-section="circle"
        fill="none"
        r="20"
        stroke-miterlimit="10"
        stroke-width="2"
      />
    </svg>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
